libraries/DataFlash/LogFile.cpp

```diff
@@ -425,8 +425,9 @@ void DataFlash_Class::Log_Write_RSSI(AP_RSSI &rssi)
     WriteBlock(&pkt, sizeof(pkt));
 }
 
-void DataFlash_Class::Log_Write_Baro_instance(AP_Baro &baro, uint64_t time_us, uint8_t baro_instance, enum LogMessages type)
+void DataFlash_Class::Log_Write_Baro_instance(uint64_t time_us, uint8_t baro_instance, enum LogMessages type)
 {
+    AP_Baro &baro = AP::baro();
```

Comment:
const?

Reply:
it calls `baro.get_climb_rate()` that uses DerivativeFilter::slope() which modifies its state. So we can't propagate constness, at least not in this PR.